src/Symfony/Component/Form/Extension/Validator/Constraints/FormValidator.php

```diff
@@ -96,11 +96,19 @@ public function validate($form, Constraint $constraint)
                     ? (string) $form->getViewData()
                     : gettype($form->getViewData());
 
-                $this->context->buildViolation($config->getOption('invalid_message'))
-                    ->setParameters(array_replace(array('{{ value }}' => $clientDataAsString), $config->getOption('invalid_message_parameters')))
+                $failure = $form->getTransformationFailure();
```

Comment:
As said in https://github.com/symfony/symfony/pull/20966, I'd like to deprecate this handling, although I'm not against this feature, this should be revised later :)